### PR TITLE
Add error flags

### PR DESCRIPTION
**Pull request by PNieck**
## Description

Add ekf error reporting

## Motivation and Context

Possibility not to ignore possible EKF errors.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: ekf/ekflib.c [hunk 1]
```diff
@@ -279,20 +284,37 @@ static void *ekf_thread(void *arg)
 
 	printf("ekf: starting ekf thread\n");
 
+	errno = 0;
```
Comment (niewim19) [inline ekf/ekflib.c:287]:
```suggestion
	errno = EOK;
```

File: ekf/ekflib.c [hunk 2]
```diff
@@ -279,20 +284,37 @@ static void *ekf_thread(void *arg)
...
 	ekf_common.run = 1;
 
 	/* Kalman loop */
-	meas_timeGet(&ekf_common.lastTime);
+	if (meas_timeGet(&ekf_common.lastTime) != 0) {
+		ekf_common.status |= errno == 0 ? EKF_MEAS_EOF : EKF_ERROR;
```
Comment (niewim19) [inline ekf/ekflib.c:292]:
```suggestion
		ekf_common.status |= (errno == EOK) ? EKF_MEAS_EOF : EKF_ERROR;
```

File: ekf/ekflib.c [hunk 3]
```diff
@@ -279,20 +284,37 @@ static void *ekf_thread(void *arg)
...
+		if (ekf_common.initVals.measSource == srcLog) {
```
Comment (niewim19) [inline ekf/ekflib.c:293]: Add a suitable comment here, please

File: ekf/ekflib.c [hunk 4]
```diff
@@ -279,20 +284,37 @@ static void *ekf_thread(void *arg)
...
+	}
 
 	lastBaroUpdate = ekf_common.lastTime;
 	lastGpsUpdate = ekf_common.lastTime;
 
 	while (ekf_common.run == 1) {
 		usleep(1000);
-		loopStep = ekf_dtGet();
+		if (ekf_dtGet(&loopStep) != 0) {
+			ekf_common.status |= errno == 0 ? EKF_MEAS_EOF : EKF_ERROR;
```
Comment (niewim19) [inline ekf/ekflib.c:304]:
```suggestion
			ekf_common.status |= (errno == EOK) ? EKF_MEAS_EOF : EKF_ERROR;
```

File: ekf/ekflib.c [hunk 5]
```diff
@@ -279,20 +284,37 @@ static void *ekf_thread(void *arg)
...
+			if (ekf_common.initVals.measSource == srcLog) {
+				break;
+			}
+		}
 
 		/* IMU polling is done regardless on update procedure */
-		meas_imuPoll(&imuTime);
+		if (meas_imuPoll(&imuTime) != 0) {
+			ekf_common.status |= errno == 0 ? EKF_MEAS_EOF : EKF_ERROR;
```
Comment (niewim19) [inline ekf/ekflib.c:312]:
```SUGGESTION
			ekf_common.status |= (errno == EOK) ? EKF_MEAS_EOF : EKF_ERROR;
```

File: ekf/ekflib.c [hunk 6]
```diff
@@ -303,6 +325,13 @@ static void *ekf_thread(void *arg)
 				updateStep = ekf_common.currTime - lastBaroUpdate;
 				lastBaroUpdate = ekf_common.currTime;
 			}
+			else {
+				ekf_common.status |= errno == 0 ? EKF_MEAS_EOF : EKF_ERROR;
```
Comment (niewim19) [inline ekf/ekflib.c:329]:
```SUGGESTION
				ekf_common.status |= (errno == EOK) ? EKF_MEAS_EOF : EKF_ERROR;
```

File: ekf/ekflib.c [hunk 7]
```diff
@@ -311,6 +340,13 @@ static void *ekf_thread(void *arg)
 				updateStep = ekf_common.currTime - lastGpsUpdate;
 				lastGpsUpdate = ekf_common.currTime;
 			}
+			else {
+				ekf_common.status |= errno == 0 ? EKF_MEAS_EOF : EKF_ERROR;
```
Comment (niewim19) [inline ekf/ekflib.c:344]:
```SUGGESTION
				ekf_common.status |= (errno == 0) ? EKF_MEAS_EOF : EKF_ERROR;
```

File: utils/devekf.c [hunk 8]
```diff
@@ -103,6 +103,26 @@ static inline void printUavAcc(ekf_state_t *uavState)
...
+	if ((uavStatus & EKF_STATUS_RUNNING) != 0) {
+		printf("EKF stopped working");
+		res = -1;
+	}
+	else if ((uavStatus & EKF_ERROR) != 0) {
+		printf("EKF error occurred\n");
+		res = -1;
+	}
+	else if ((uavStatus & EKF_MEAS_EOF) != 0) {
+		printf("EKF: meas module encountered EOF");
```
Comment (niewim19) [inline utils/devekf.c:119]: Why we do not halt `devekf` on EOF?

**Review by niewim19 — CHANGES_REQUESTED** [review]
Looks good. General remarks:
 - if `errno` is extensively used, we can signal _all good_ with EOK, as it is part of errno interface,
 - let's think about single ekf breking point - currently each data polling can break the ekf loop (stop it) separately. As that loop has clearly _data getting_ part and _math part_ it may be possible to poll whatever is needed and break the loop in between these parts if necessary to de-clutter the code out of `return`s.

File: ekf/logs/reader.c [hunk 9]
```diff
@@ -83,21 +85,27 @@ static int ekflog_nextLogSeek(char logIndicator)
 
 int ekflog_timeRead(time_t *timestamp)
 {
+	errno = 0;
```
[the remark this answers is not in the document]
Comment (PNieck) [reply ekf/logs/reader.c:88]: `EOK` is not available on host

**Comment by PNieck** [comment]
I cannot use `EOK`, because it is not available on host

**Comment by niewim19** [comment]
> I cannot use `EOK`, because it is not available on host

Ok, if so then `0` is ok.

File: ekf/ekflib.c [hunk 10]
```diff
@@ -261,24 +280,45 @@ static void *ekf_thread(void *arg)
 	time_t imuTime;
 	time_t loopStep, updateStep;
 	update_engine_t *currUpdate;
+	bool dataAcqSucc;
```
Comment (niewim19) [inline ekf/ekflib.c:283]: Slight modification for more appropriate name ;)
```suggestion
	bool dataAcqOk;
```
On the other hand `ekf_common.run` may be used for this purpose: just set this variable to false in `*Acq()` ladder and then before math part there is check if we still want to run the loop.

File: ekf/ekflib.h [hunk 11]
```diff
@@ -21,6 +21,8 @@
 
 /* Ekf status flags */
 #define EKF_STATUS_RUNNING (1 << 0)
+#define EKF_ERROR          (1 << 1)
```
Comment (niewim19) [inline ekf/ekflib.h:24]:
```suggestion
#define EKF_ERROR          (1 << 1) /* general purpose error flag */
```

**Review by niewim19 — CHANGES_REQUESTED** [review]

**Review by niewim19 — APPROVED** [review]

**Pull request merged**